### DISCUSSION TITLE
multhree.c doubt (TLE)

### DISCUSSION THREAD

**Discussion by spycedcoffee** [Nimesh-Srivastava/DSA#41] [Q&A] [answered]
[Link to question](https://www.codechef.com/LRNDSA01/problems/MULTHREE)

Code :- <br>
```
#include<stdio.h>

int main()
{
	int t, k, d0, d1;
	scanf("%d", &t);
	while(t--)
	{
		int i = 2, sum = 0, j;
		scanf("%d %d %d", &k, &d0, &d1);
		const int s = 1e13;
		int a[s];
		a[0] = d0;
		a[1] = d1;
		sum = a[0] + a[1];
		for (j = 0; j < k-2; j++)
		{
			a[i] = sum%10;
			sum = sum + a[i];
			i++;
		}
		if(sum%3==0)
		{
			printf("YES\n");
		}
		else
		{
			printf("NO\n");
		}
	}
}
```

this is my code for the given question. On submission it says time limit exceeded even though output is correct.
what am I doing wrong?

**Comment by Nimesh-Srivastava** [comment]
Have you tried using `long long int` instead of `int` for storing the sum as well as the value of `k`?

**Comment by Nimesh-Srivastava** [comment, marked as answer]
Solved this in C++. Changing the data type is necessary, but is not enough. For large values of k, the loop to find the digits is going to run many times (maybe even upto 10^12), which is not ideal. You need to find a way to cut down the number of loop runs.

Best possible way is to do multiple calculations in a single run, e.g., calculating the next 4 digits at once.

I will post the solution soon. Try it for yourself till then.

**Reply by spycedcoffee** [reply]
oh okay, got it! thanks
